jobbergate-api/jobbergate_api/apps/models.py

```diff
@@ -58,10 +60,22 @@ class IdMixin:
 
     Attributes:
         id: The id of the job script template.
+        cloned_from: Specify the id of the row that this row was cloned from.
     """
 
     id: Mapped[int] = mapped_column(Integer, primary_key=True)
 
+    @declared_attr
+    def cloned_from_id(cls) -> Mapped[int | None]:
```

Comment:
Why is this specified as a `declared_attr`?

Reply:
[SQLAlchemy requires](https://docs.sqlalchemy.org/en/20/orm/declarative_mixins.html) it in a mixing class so that we can refer to another column from it. Otherwise, the error message is very specific:

```text
sqlalchemy.exc.InvalidRequestError:
Columns with foreign keys to other columns must be declared as @declared_attr callables on declarative mixin classes.
```

Reply:
Oh, gotcha. Nice.